### PR TITLE
Kron

### LINKED ISSUE #12553: kron

### PR DESCRIPTION
**Pull request by ipriyam26**
closes #12553

### CODE REVIEW

**Review by fspyridakos — COMMENTED** [review]
Just a small comment about the test, function looks good!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 1]
```diff
@@ -1084,3 +1084,38 @@ def test_diff(
         prepend=prepend[0],
         append=append[0],
     )
+
+
+@handle_frontend_test(
+    fn_tree="torch.kron",
+    dtype_input_other_dim=dtype_value1_value2_axis(
```
[the remark this answers is not in the document]
Comment (ipriyam26) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1091]: I was looking at another function and just thought that this is giving out two arrays so I used it no specific reason, Here I have removed it for simple dtype and values in next commit

File: ivy/functional/frontends/torch/miscellaneous_ops.py [hunk 2]
```diff
@@ -330,3 +330,8 @@ def diff(input, n=1, dim=-1, prepend=None, append=None):
 @to_ivy_arrays_and_back
 def broadcast_shapes(*shapes):
     return ivy.broadcast_shapes(*shapes)
+
+
+@to_ivy_arrays_and_back
+def kron(input, other, *, out=None):
+    return ivy.kron(input, other, out=out)
```
Comment (fspyridakos) [inline ivy/functional/frontends/torch/miscellaneous_ops.py:337]: add `promote_types_of_torch_inputs` to handle promotion

File: ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py [hunk 3]
```diff
@@ -1115,3 +1115,35 @@ def test_torch_broadcast_shapes(
...
+@handle_frontend_test(
+    fn_tree="torch.kron",
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=1,
+        max_dim_size=10,
+        num_arrays=2,
+        shared_dtype=True,
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_miscellaneous_ops.py:1129]: remove `shared_dtype`

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Looks pretty good, just one final thing, probably worth doing a merge with upstream since I changed some stuff in the file as other functions were using the wrong promotion. You should use `promote_types_of_torch_inputs`

**Pull request closed without merging**